### ISSUE TITLE
Split twisted.test.test_reflect.FullyQualifiedNameTests.test_method in two

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5331]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5331|
|Type|enhancement|
|Created|2011-10-22 16:13:26Z|

Splitting this test in two tests, one for bound methods and one for unbound methods, allows skipping the unbound method test under Python 3. Patch attached.

Attachments:

* [split_test_method.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/494/494b558646e128f6319a38e8691e7cc55c7d8165/87578eae59b00d9146cc81cdd2e846f702aafe56.patch) (1247 bytes) - added by antoine on 2011-10-22 16:13:35Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5331 5331
type__enhancement enhancement
reporter__antoine antoine
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1319300006000000 1319300006000000
changetime__1320717093000000 1320717093000000
version__None None
owner__exarkun exarkun
cc__antoine
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun|
|-|-|

I like the idea of splitting this test in two, because it's testing two difference features.  One unit test should cover one feature.  I'm not terribly interested in the ability to skip one of the tests on Python 3 because that seems like a sad degradation of functionality. :)  The patch doesn't propose skipping anything now though, so that's easy!

The test method names should follow the `test_fooBar` convention rather than the `test_foo_bar` convention.  Otherwise looks great.  I'll tweak that and apply the patch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [33095]) Split the unit test for bound and unbound method fully qualified names into two tests.

Author: antoine
Reviewer: exarkun
Fixes: [#5331](https://github.com/twisted/twisted/issues/5331)

The bound method and unbound method cases of `fullyQualifiedName`
were previously tested in a single test method.  Split that test
method into two, one for each case.
